bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -974,6 +974,9 @@ private static boolean checkIsArrayType(BType sourceType, BArrayType targetType,
             sourceArrayType = (BArrayType) sourceType;
         } else {
             BTupleType sourceTupleType = (BTupleType) sourceType;
+            if (sourceTupleType.getTupleTypes().isEmpty()) {
```

Comment:
Wouldn't this cause issues when the source type has just a rest-descriptor?

I think the following will produce an incorrect result with this change.

```ballerina
import ballerina/io;

public function main() {
    [int...] x = [1, 2];
    any y = x;
    io:println(y is string[]);
}
```

We should ideally have tests for the same too.

Reply:
Thanks fixed